src/CommunityToolkit.Maui/Views/RatingView/RatingView.shared.cs

```diff
@@ -0,0 +1,748 @@
+		AddChildrenToControl(0, MaximumRating);
+	}
+
+	/// <summary>Occurs when <see cref="Rating"/> is changed.</summary>
+	public event EventHandler<RatingChangedEventArgs> RatingChanged
+	{
+		add => weakEventManager.AddEventHandler(value);
+		remove => weakEventManager.RemoveEventHandler(value);
+	}
+
+	///<summary>Defines the shape to be drawn.</summary>
+	public string CustomItemShape
+	{
+		get => (string)GetValue(CustomItemShapeProperty);
+		set => SetValue(CustomItemShapeProperty, value);
+	}
+
+	/// <summary>Gets or sets a value of the empty rating color property.</summary>
+	[AllowNull]
+	public Color EmptyColor
+	{
+		get => (Color)GetValue(EmptyColorProperty);
+		set => SetValue(EmptyColorProperty, value ?? Colors.Transparent);
+	}
+
+	/// <summary>Gets or sets a value of the filled rating color property.</summary>
+	[AllowNull]
+	public Color FilledColor
+	{
+		get => (Color)GetValue(FilledColorProperty);
+		set => SetValue(FilledColorProperty, value ?? Colors.Transparent);
+	}
+
+	///<summary>Gets or sets a value indicating if the controls is read-only.</summary>
+	public bool IsReadOnly
+	{
+		get => (bool)GetValue(IsReadOnlyProperty);
+		set => SetValue(IsReadOnlyProperty, value);
+	}
+
+	/// <summary>Gets or sets a value indicating the padding between the rating item and the rating shape.</summary>
+	public Thickness ItemPadding
+	{
+		get => (Thickness)GetValue(ItemPaddingProperty);
+		set => SetValue(ItemPaddingProperty, value);
+	}
+
+	/// <summary>Gets or sets a value indicating the shape size.</summary>
+	public double ItemShapeSize
+	{
+		get => (double)GetValue(ItemShapeSizeProperty);
+		set => SetValue(ItemShapeSizeProperty, value);
+	}
+
+	/// <summary>Gets or sets a value indicating the maximum rating.</summary>
+	public int MaximumRating
+	{
+		get => (int)GetValue(MaximumRatingProperty);
+		set
+		{
+			switch (value)
+			{
+				case <= 0:
+					throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(MaximumRating)} must be greater than 0");
+				case > RatingViewDefaults.MaximumRatingLimit:
+					throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(MaximumRating)} cannot be greater than {nameof(RatingViewDefaults.MaximumRatingLimit)}");
+				default:
+					SetValue(MaximumRatingProperty, value);
+					break;
+			}
+		}
+	}
+
+	/// <summary>Gets or sets a value indicating the rating.</summary>
+	public double Rating
+	{
+		get => (double)GetValue(RatingProperty);
+		set
+		{
+			if (value < 0)
+			{
+				throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(Rating)} cannot be less than 0");
+			}
+			if (value > MaximumRating)
+			{
+				throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(Rating)} cannot be greater than {nameof(MaximumRating)}");
+			}
+			
+			SetValue(RatingProperty, value);
+		}
+	}
+
+	/// <summary>Command that is triggered when the value of <see cref="Rating"/> is changed.</summary>
+	public ICommand? RatingChangedCommand
```

Comment:
What's the thinking behind having a command here? Can't the value just be data bound to handle it changing?